src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/EgressTests.cs

```diff
@@ -211,6 +211,39 @@ public async Task SharedConcurrencyLimitTest()
                 });
         }
 
+        /// <summary>
+        /// Tests that turning off HTTP egress results in an error for dumps and logs (gcdumps and traces are currently not tested)
+        /// </summary>
+        [Fact]
+        public async Task DisableHttpEgressTest()
+        {
+            await ScenarioRunner.SingleTarget(
+                _outputHelper,
+                _httpClientFactory,
+                DiagnosticPortConnectionMode.Connect,
+                TestAppScenarios.AsyncWait.Name,
+                appValidate: async (appRunner, appClient) =>
+                {
+                    ProcessInfo processInfo = await appClient.GetProcessAsync(appRunner.ProcessId);
+                    Assert.NotNull(processInfo);
+
+                    // Dump Error Check
+                    ValidationProblemDetailsException validationProblemDetailsExceptionDumps = await Assert.ThrowsAsync<ValidationProblemDetailsException>(
+                        () => appClient.CaptureDumpAsync(appRunner.ProcessId, DumpType.Mini));
+                    Assert.Equal(HttpStatusCode.BadRequest, validationProblemDetailsExceptionDumps.StatusCode);
+                    Assert.Equal(StatusCodes.Status400BadRequest, validationProblemDetailsExceptionDumps.Details.Status);
```

Comment:
Given that the error message doesn't change between the endpoints or what is calling it, do you think you could add an assert for the expected error message? Maybe you can reference the resource string instead of embedding the literal message.